src/github.com/getlantern/flashlight/client/pluggable_transports.go

```diff
@@ -53,10 +53,11 @@ func defaultDialFactory(s *ChainedServerInfo, deviceID string) (dialFN, error) {
 			op := ops.Begin("dial_to_chained").ChainedProxy(s.Addr, "https")
 			defer op.End()
 
-			conn, err := tlsdialer.Dial("tcp", addr, false, &tls.Config{
-				ClientSessionCache: sessionCache,
-				InsecureSkipVerify: true,
-			})
+			conn, err := tlsdialer.DialTimeout(netx.DialTimeout, chainedDialTimeout,
```

Comment:
As a separate package, 1 min should be good enough. [Windows has a 72s timeout initially](http://serverfault.com/questions/193160/which-is-the-default-tcp-connect-timeout-in-windows).